src/main/java/org/mockbukkit/mockbukkit/entity/EnderDragonMock.java

```diff
@@ -0,0 +1,190 @@
+package org.mockbukkit.mockbukkit.entity;
+
+import io.papermc.paper.math.Position;
+import org.bukkit.boss.BossBar;
+import org.bukkit.entity.EnderCrystal;
+import org.jetbrains.annotations.Unmodifiable;
+import org.mockbukkit.mockbukkit.ServerMock;
+import org.bukkit.Location;
+import org.bukkit.boss.DragonBattle;
+import org.bukkit.entity.ComplexEntityPart;
+import org.bukkit.entity.EnderDragon;
+import org.bukkit.entity.EntityType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.mockbukkit.mockbukkit.exception.UnimplementedOperationException;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+public class EnderDragonMock extends AbstractBossMock implements EnderDragon
+{
+
+	private Location podium;
+	private Phase phase = Phase.HOVER;
+	private final DragonBattle battle;
+
+	/**
+	 * Constructs a new {@link MonsterMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public EnderDragonMock(@NotNull ServerMock server, @NotNull UUID uuid)
```

Comment:
Most of these methods in this anonymous class are not really implemented, yes they don't throw unimplemented operation exception, but their behavior does follow paper behavior.

Either implement them or don't

Reply:
Where can i implement a class to handle it properly instead of creating an anonymius object on the fly?

Reply:
> Where can i implement a class to handle it properly instead of creating an anonymius object on the fly?

You can just make an implementation in the entity package, just add mock to the end of the name, and we're set 😄